tdmq/client/client.py

```diff
@@ -0,0 +1,54 @@
+import requests
+import tiledb
+import numpy as np
+
+from tdmq.client.sensors import NonScalarSensor
+
+sensor_classes = {
+    'meteoRadar': NonScalarSensor,
+}
+
+
+class Client:
+    def __init__(self, tdmq_base_url, tiledb_ctx=None):
+        self.base_url = tdmq_base_url
+        self.tiledb_ctx = tiledb_ctx
+        self.sensor_types = None
+        self.update_sensor_types()
+
+    def update_sensor_types(self):
+        stypes = requests.get(f'{self.base_url}/sensor_types').json()
+        self.sensor_types = dict((st['name'], st) for st in stypes)
+
+    def get_sensors(self, args):
+        res = requests.get(f'{self.base_url}/sensors', params=args).json()
+        return [self.get_sensor_proxy(r['code']) for r in res]
+
+    def get_sensor_proxy(self, code):
+        res = requests.get(f'{self.base_url}/sensors/{code}').json()
+        assert res['code'] == code
+        # FIXME we need to fix this 'type' thing
+        stype = self.sensor_types[res['type']]
+        return sensor_classes[stype['type']](
+            self, code, stype, res)
+
+    def get_timeseries(self, code, args):
+        return requests.get(f'{self.base_url}/sensors/{code}/timeseries',
```

Comment:
I think this is an inconsistent interface. get_sensors method returns a Sensor Object, while get_timeseries returns a JSON. I suggest the client should only return JSON, while Sensor subclasses has the logic for building theirselves from the JSON